geoh5py/objects/drillhole.py

```diff
@@ -408,7 +408,10 @@ def add_data(
                 )
 
             attributes, new_property_group = self.validate_data(
-                attributes, property_group, collocation_distance=collocation_distance
+                attributes,
+                property_group,
+                collocation_distance=collocation_distance,
+                type_=attributes.get("type"),

```

Comment:
Can't we check the type directly in the collocation_distance method?

Reply:
collocation_distance isn't a float?
type is a string....
What if both are passed?